src/demo/OceanObject.js

```diff
@@ -0,0 +1,210 @@
+import {fishData, FishBodyPart} from '../utils/fishData';
+import constants from './constants';
+import {
+  bodyAnchorFromType,
+  colorForFishPart,
+  randomInt,
+  clamp
+} from './helpers';
+import _ from 'lodash';
+
+let fishPartImages = {};
+let intermediateCanvas;
+let intermediateCtx;
+// Load a single fish part image.
+const loadFishPartImage = data => {
+  return new Promise((resolve, reject) => {
+    const img = new Image();
+    img.addEventListener('load', e => resolve({img, data}));
+    img.addEventListener('error', () => {
+      reject(new Error(`failed to load image at ${data.src}`));
+    });
+    img.src = data.src;
+  });
+};
+
+// Load all fish part images, and store them in fishPartImages.
+export const loadAllFishPartImages = () => {
+  intermediateCanvas = document.createElement('canvas');
+  intermediateCtx = intermediateCanvas.getContext('2d');
+  intermediateCanvas.width = constants.fishCanvasWidth;
+  intermediateCanvas.height = constants.fishCanvasHeight;
+
+  let fishPartImagesToLoad = [];
+  Object.keys(fishData)
+    .filter(partName => partName !== 'colorPalettes')
+    .forEach((partName, partIndex) => {
+      Object.keys(fishData[partName]).forEach(variationName => {
+        const partData = {
+          partIndex: fishData[partName][variationName].type,
+          variationIndex: fishData[partName][variationName].index,
+          src: fishData[partName][variationName].src
+        };
+        fishPartImagesToLoad.push(partData);
+      });
+    });
+
+  return Promise.all(fishPartImagesToLoad.map(loadFishPartImage)).then(
+    results => {
+      results.forEach(result => {
+        if (fishPartImages[result.data.partIndex] === undefined) {
+          fishPartImages[result.data.partIndex] = {};
+        }
+        fishPartImages[result.data.partIndex][result.data.variationIndex] =
+          result.img;
+      });
+    }
+  );
+};
+
+export const generateOceanObject = (allowedClasses, id) => {
+  const idx = Math.floor(Math.random() * allowedClasses.length);
+  const newOceanObject = new allowedClasses[idx](id);
+  newOceanObject.randomize();
+  return newOceanObject;
+};
+
+export class OceanObject {
+  constructor(id) {
+    this.id = id;
+    this.knnData = null;
+    this.result = null;
+  }
+  randomize() {}
```

Comment:
should these unimplemented methods throw an error like:

```
raise new Error("Not yet implemented");
```

so that we know if subclasses are calling these methods without their own implementations?

Reply:
ah yes. I meant to go back and do this!

Added `throw 'Not yet implemented!'` errors to both methods.